plugins/out_kinesis_streams/kinesis_api.c

```diff
@@ -97,6 +97,61 @@ static int init_put_payload(struct flb_kinesis *ctx, struct flush *buf,
     return -1;
 }
 
+/*
+ * Count the substring number in string.
+ */
+static int get_substring_number(char *str1, char *str2){
+    int x = 0;
+    char *p;
+
+        do{
+        p = strstr(str1, str2);
+        if (p != NULL) {
+            str1 = p + 1;
+            x = x + 1;
+        }
+    }
+    while(p != NULL);
+    return x;
+}
+
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
+static void get_time_format_string(char *str, size_t maxsize,
```

Comment:
We will use this function in the kinesis_streams code as well? Then I think it should be in the flb_aws_util code base.

Reply:
Notice that in the original code, we can fail to write to the request buffer: https://github.com/fluent/fluent-bit/blob/master/plugins/out_kinesis_firehose/firehose_api.c#L239

So I think we need a return value here.

Reply:
char **out_buf;

Reply:
*out_buf = new_memory;

size_t *out_size;

*out_size = size;

Reply:
flb_sds_t *out_buf;